designer/server/src/main/scala/pl/touk/nussknacker/ui/component/ComponentsUsageHelper.scala

```diff
@@ -12,21 +13,24 @@ object ComponentsUsageHelper {
 
   def computeComponentsUsageCount(
       processesDetails: List[ScenarioWithDetailsEntity[ScenarioComponentsUsages]],
-      processingTypeAndComponentInfoToComponentId: (ProcessingType, ComponentInfo) => ComponentId,
+      processingTypeToNonFragmentComponents: Map[ProcessingType, Map[ComponentInfo, ComponentStaticDefinition]]
   ): Map[ComponentId, Long] = {
-    computeComponentsUsage(processesDetails, processingTypeAndComponentInfoToComponentId)
+    computeComponentsUsage(processesDetails, processingTypeToNonFragmentComponents)
       .mapValuesNow(usages => usages.map { case (_, nodeIds) => nodeIds.size }.sum)
   }
 
   def computeComponentsUsage(
       processesDetails: List[ScenarioWithDetailsEntity[ScenarioComponentsUsages]],
-      processingTypeAndComponentInfoToComponentId: (ProcessingType, ComponentInfo) => ComponentId,
+      processingTypeToNonFragmentComponents: Map[ProcessingType, Map[ComponentInfo, ComponentStaticDefinition]]
   ): Map[ComponentId, List[(ScenarioWithDetailsEntity[_], List[NodeUsageData])]] = {
     def flattenUsages(processesDetails: List[ScenarioWithDetailsEntity[ScenarioComponentsUsages]]) = for {
       processDetails    <- processesDetails
       componentInfoNode <- processDetails.json.value.toList
       (componentInfo, nodeIds) = componentInfoNode
-      componentId = processingTypeAndComponentInfoToComponentId(processDetails.processingType, componentInfo)
+      componentId = processingTypeToNonFragmentComponents(processDetails.processingType)
+        .get(componentInfo)
+        .map(_.componentIdUnsafe)
+        .getOrElse(ComponentId.default(processDetails.processingType, componentInfo))
```

Comment:
Please write a comment that we pick the default one instead of skipping this usage because for fragments, there won't be an id in the `processingTypeToNonFragmentComponents`